components/form/TaskForm.tsx

```diff
@@ -32,39 +46,42 @@ const validateTaskFields = (fields: TaskFields, projectId: number): Task => {
     throw Error("Invalid task form: missing required fields")
 }
 
-export function CreateTaskForm({
+function TaskForm({
+    task: taskOrNull,
     project,
     projectId,
-    afterSubmit,
+    onSubmit,
     onCancel,
 }: TaskFormProps): JSX.Element {
-    const [taskFields, setTaskFields] = useState<TaskFields>({
-        project,
-        billable: true,
-    })
-    const { postTask } = useUpdateTasks()
+    const [taskFields, setTaskFields] = useState<TaskFields>(
+        taskOrNull || {
+            project,
+            billable: true,
+        }
+    )
 
     const [errorMessage, setErrorMessage] = useState<string>("")
     const errorHandler = (error: Error) => setErrorMessage(`${error}`)
 
-    const onSubmit = async (event: React.MouseEvent) => {
+    const abortSubmission = onCancel && onCancel
+
+    const formOnSubmit = (event: React.MouseEvent) => {
         event.preventDefault()
         try {
             const task = validateTaskFields(taskFields, projectId)
-            const newTask = await postTask(task, errorHandler)
-            return afterSubmit && afterSubmit(newTask)
+            onSubmit(task)
         } catch (error) {
             errorHandler(error as Error)
-            return null
         }
     }
 
     return (
-        <>
+        <FormBackground>
```

Comment:
I'd prefer this to be called `FormContainer` and the current `FormContainer` to be called something like `FormFieldsContainer`.

Suggestion:

1. Navigate to the current `FormContainer` below on VSCode.
2. Press `f12` to goto definition.
3. Press `f2` to rename.
4. Navigate to the definition of `FormBackground`. Press `f2` to rename.